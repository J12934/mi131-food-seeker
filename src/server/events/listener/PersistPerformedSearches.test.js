const {
    SearchEventEmitter,
    PersistPerformedSearches,
} = require('./PersistPerformedSearches');

const mongoose = require('../../database/connection');
const Search = require('../../database/models/Search');

beforeEach(() => {
    return Search.remove({});
});

afterAll(() => {
    return mongoose.disconnect();
});

describe('SearchEventListener', () => {
    it('save query to database on search started event', async () => {
        await PersistPerformedSearches({
            term: 'Burger',
            coordinates: { latitude: 54, longitude: 11 },
        });
        const searches = await Search.find({}).select({ __v: 0, _id: 0 });

        expect(searches).toMatchSnapshot();
    });

    it('registered to SearchEventEmitter', async () => {
        expect(SearchEventEmitter.eventNames()).toContain(
            SearchEventEmitter.types.SEARCH_STARTED
        );
    });
});
