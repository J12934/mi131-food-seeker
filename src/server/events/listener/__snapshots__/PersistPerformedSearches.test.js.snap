// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEventListener save query to database on search started event 1`] = `
Array [
  Object {
    "location": Object {
      "coordinates": Array [
        11,
        54,
      ],
      "type": "Point",
    },
    "term": "Burger",
  },
]
`;
